sechub-pds-solutions/multi/docker/Multi-Ubuntu.dockerfile

```
@@ -0,0 +1,78 @@
+# SPDX-License-Identifier: MIT
+
+# The image argument needs to be placed on top
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE}
+
+# The remaining arguments need to be placed after the `FROM`
+# See: https://ryandaniels.ca/blog/docker-dockerfile-arg-from-arg-trouble/
+
+# Folders
+ARG PDS_FOLDER="/pds"
+ARG SCRIPT_FOLDER="/scripts"
+ENV TOOL_FOLDER="/tools"
+ARG WORKSPACE="/workspace"
+ENV DOWNLOAD_FOLDER="/downloads"
+ENV MOCK_FOLDER="$SCRIPT_FOLDER/mocks"
+
+# PDS
+ENV PDS_VERSION=0.25.0
+ARG PDS_CHECKSUM="ecc69561109ee98a57a087fd9e6a4980a38ac72d07467d6c69579c83c16b3255"
+
+# Shared volumes
+ENV SHARED_VOLUMES="/shared_volumes"
+ENV SHARED_VOLUME_UPLOAD_DIR="$SHARED_VOLUMES/uploads"
+
+# Create tool, pds, shared volume and download folder
+RUN  mkdir --parents "$TOOL_FOLDER" "$DOWNLOAD_FOLDER" "$PDS_FOLDER" "$SHARED_VOLUME_UPLOAD_DIR" "$WORKSPACE"
+
+# non-root user
+# using fixed group and user ids
+RUN groupadd --gid 2323 pds \
+     && useradd --uid 2323 --no-log-init --create-home --gid pds pds
+
+# Update image and install dependencies
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get update && \
+    apt-get --assume-yes upgrade  && \
+    apt-get --assume-yes install wget openjdk-11-jre-headless pip && \
+    apt-get --assume-yes clean
+
+# Copy scripts
+COPY scripts $SCRIPT_FOLDER
+RUN chmod -R +x $SCRIPT_FOLDER
+
+# Mock folder
+COPY mocks $SCRIPT_FOLDER/mocks/
+
+# Install Flawfinder, Bandit, njsscan and mobsfscan
+COPY packages.txt $TOOL_FOLDER/packages.txt
+RUN pip install -r $TOOL_FOLDER/packages.txt
+
+# Install the SecHub Product Delegation Server (PDS)
+RUN cd "$PDS_FOLDER" && \
+    # download checksum file
+    wget --no-verbose "https://github.com/Daimler/sechub/releases/download/v$PDS_VERSION-pds/sechub-pds-$PDS_VERSION.jar.sha256sum" && \
+    # download pds
+    wget --no-verbose "https://github.com/Daimler/sechub/releases/download/v$PDS_VERSION-pds/sechub-pds-$PDS_VERSION.jar" && \
+    # verify that the checksum and the checksum of the file are same
+    sha256sum --check sechub-pds-$PDS_VERSION.jar.sha256sum
+
+# Copy PDS configfile
+COPY pds-config.json /$PDS_FOLDER/pds-config.json
+
+# Copy run script into container
+COPY run.sh /run.sh
+RUN chmod +x /run.sh
+
+# Create the PDS workspace
+WORKDIR "$WORKSPACE"
+
+# Change owner of tool, workspace and pds folder as well as /run.sh
+RUN chown --recursive pds:pds $TOOL_FOLDER $SCRIPT_FOLDER $DOWNLOAD_FOLDER $WORKSPACE $PDS_FOLDER $SHARED_VOLUMES /run.sh
```

```suggestion
RUN chown --recursive pds:pds $TOOL_FOLDER $SCRIPT_FOLDER $DOWNLOAD_FOLDER $WORKSPACE  $SHARED_VOLUMES
```
User pds needs no write permission in $PDS_FOLDER and /run.sh.
It is handy while developing, but a productive environment should be as safe as possible.

---

This only is to change the ownership to the pds user. If I want to set read and execute rights I would have to use `chmod`.

---

@Jeeppler : The ownership is the crucial thing.
Imagine, a hacker manages to log in to the container. The shell will run under `pds` user,
File changes and `chmod` can be executed only if the file/directory is owned by `pds` (or where it has write permissions).

So if a directory belongs to root, then it cannot be altered by the hacker.
As a consequence, the less directories belong to `pds`, the more secure we are.
-> imho only directories where the application *really* needs write permissions should be owned by `pds`.

---

Thanks implemented it for PDS + GoSec and PDS + Multi.